### DISCUSSION TITLE
105. MySQL(InnoDB)에서 Primary Key(PK)가 없을 때 생성되는 클러스터형 인덱스가 성능에 어떤 영향을 미치나요?

### DISCUSSION THREAD

**Discussion by jongmee** [jewelry-box/cs-interview#109] [General]
[관련 질문]
[58. MySQL에서 테이블에 기본 키(Primary Key)가 없는 경우, 클러스터드 인덱스는 어떻게 설정되나요?](https://github.com/jewelry-box/cs-interview/discussions/60)

**Comment by cutehumanS2** [comment]
1. **쿼리 최적화에 제약이 생길 수 있습니다.**
    Row ID 같이 자동 생성된 PK는 사용자에게 노출되지 않아 쿼리 조건에서 직접 활용할 수 없습니다. 따라서 개발자는 데이터 구조를 명확히 이해하지 못하거나, JOIN, WHERE 조건, ORDER BY 등의 쿼리 최적화에 제약이 생길 수 있습니다.
    
2. **범위 조회와 정렬 성능이 저하됩니다.**
    클러스터링 키는 데이터의 물리적 정렬 순서를 결정하는데, 의미있는 PK가 아닌 자동 생성된 Row ID를 클러스터링 키로 사용하면 데이터 정렬이 무의미해집니다. 따라서 범위 조회나 정렬 작업에서 효율적인 데이터 접근이 불가능해집니다. 
    

<br>
[참고한 자료]
책 <Real MySQL 8.0 (1권)> - 8.8 클러스터링 인덱스
